### PR TITLE
Make Transforms into classes and convert simplex to stick breaking transform

### PR DESCRIPTION
**Pull request by jsalvatier**
Avoids pickling errors. 

I also wonder if we should change the name of transforms to be `log`, `logodds`, `simplex`. But maybe just "log" would be too confusing? 

Maybe we should just import the module transform and not the contents? Or maybe to be pythonic we should call them `log_transform` etc?

### CODE REVIEW

**Comment by jsalvatier** [comment]
Also curious if people think there's a better logic for `x = Uniform('x', 3, 10, transform=interval_transform(3,10))`. Would be kind of nice to just be able to say `transform=interval_transform`. Should a transform have access to the distribution itself?

**Comment by jsalvatier** [comment]
I've now changed the simplex transform to use the stick breaking transform.

**Comment by jsalvatier** [comment]
@twiecki can you take a look at this branch? 

(I still have to figure out how to make the tests take less resources so they pass consistently).

**Comment by twiecki** [comment]
@jsalvatier I'll try to take a look soon. Exciting that you have the stick-breaking transform working.

I actually didn't realize that when I merged @aflaxman's PR that it reverted parallel nosetests. We might want to put that in again.

File: pymc3/distributions/continuous.py [hunk 1]
```diff
@@ -18,12 +18,12 @@
 
 class PositiveContinuous(Continuous):
     """Base class for positive continuous distributions"""
-    def __init__(self, transform=logtransform, *args, **kwargs):
+    def __init__(self, transform=transforms.log, *args, **kwargs):
```
Comment (twiecki) [inline pymc3/distributions/continuous.py:21]: I like the change to use the namespace.

File: pymc3/distributions/transforms.py [hunk 2]
```diff
@@ -47,46 +57,118 @@ def __init__(self, dist, transform, *args, **kwargs):
                 testval, dist.defaults, 
                 *args, **kwargs)
 
-
     def logp(self, x):
         return self.dist.logp(self.transform_used.backward(x)) + self.transform_used.jacobian_det(x)
 
 transform = Transform
 
+class Log(ElemwiseTransform):
```
Comment (twiecki) [inline pymc3/distributions/transforms.py:65]: Having these as explicit classes also makes things much more readable.

File: pymc3/distributions/transforms.py [hunk 3]
```diff
@@ -47,46 +57,118 @@ def __init__(self, dist, transform, *args, **kwargs):
...
+
+    def backward(self, x):
+        return exp(x)
 
-logtransform = transform("log", log, exp, idfn)
+    def forward(self, x):
+        return t.log(x)
 
+log = Log()
+logtransform = log
```
Comment (twiecki) [inline pymc3/distributions/transforms.py]: This is for backwards compat? Maybe it's better to be ruthless here, especially early on.

File: pymc3/distributions/transforms.py [hunk 4]
```diff
@@ -47,46 +57,118 @@ def __init__(self, dist, transform, *args, **kwargs):
...
-        ex = exp(-x)
-        jac = log(ex*(b-a)/(ex + 1)**2)
-        return jac
 
-    return transform("interval", interval_real, real_interval, real_interval_jacobian)
+class Interval(ElemwiseTransform):
+    name = "interval"
 
+    def __init__(self, a,b):
+        self.a=a
```
Comment (twiecki) [inline pymc3/distributions/transforms.py]: `self.a = a` for PEP8.

File: pymc3/distributions/transforms.py [hunk 5]
```diff
@@ -47,46 +57,118 @@ def __init__(self, dist, transform, *args, **kwargs):
...
+        a, b = self.a, self.b
+        r = t.log((x - a) / (b - x))
+        return r
 
-simplextransform = transform("simplex",
-                             lambda p: p[:-1],
-                             lambda p: concatenate(
-                             [p, 1 - sum(p, keepdims=True)]),
-                             lambda p: constant([0]))
+def interval(a,b):
```
Comment (twiecki) [inline pymc3/distributions/transforms.py]: `a, b`

File: pymc3/distributions/transforms.py [hunk 6]
```diff
@@ -47,46 +57,118 @@ def __init__(self, dist, transform, *args, **kwargs):
...
+
+    def forward(self, x):  
+        return x[:-1]
+
+    def jacobian_det(self, x):
+        return 0
+
+sum_to_1 = SumTo1()
+
+class StickBreaking(Transform):
```
Comment (twiecki) [inline pymc3/distributions/transforms.py:139]: Maybe a docstring would be good here.

**Comment by twiecki** [comment]
This looks great. See inline comments. In general, adding more comments and docs would be good.

**Comment by aflaxman** [comment]
@twiecki sorry about that, I didn't realize that parallel tests were needed for reliable travis-ci results.  From travis docs, it seems that as long as there is some output every 10 minutes, it will keep things running: http://docs.travis-ci.com/user/build-timeouts/  I'd recommend this instead of your parallel build approach, since that way is error-prone, e.g. in the version I changes, the tests of your `glm` stuff had been inadvertently dropped.

File: pymc3/distributions/transforms.py (hunk 3 above)
Comment (jsalvatier) [inline pymc3/distributions/transforms.py]: That was my thinking. I can remove it.

**Comment by jsalvatier** [comment]
@aflaxman, tests time out when they run a total of 50 minutes+.

**Comment by jsalvatier** [comment]
@twiecki let me know if you have any other comments

**Comment by twiecki** [comment]
Looks good :+1:

**Comment by twiecki** [comment]
Is travis timing out?

**Pull request closed without merging**